### PR TITLE
fix: capture beforeAll hook errors

### LINKED ISSUE #280: propagate errors from `beforeAll` and `afterAll` hooks
+ Now error in the `beforeAll` and `afterAll` hooks would be captured and reported as error, but they will not be associated with the reporters in the correct way. 
+ Without associating these errors, the Uptime UI will have no information about what happened during the context of a single execution. 

We have to figure out a way to solve this problem.

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix #280 
+ We capture the beforeAll hook errors and any error that happens in any one of the hooks would be captured and all journeys that run on the current invocation will report that error event with `failed` status to be able to captured as part of the Uptime UI. 
+ `afterAll` errors behaves the same - we report them in the stderror logs which will be captured by Heartbeat.

### CODE REVIEW

File: __tests__/core/runner.test.ts [hunk 1]
```diff
@@ -147,6 +147,21 @@ describe('runner', () => {
...
+    const error = new Error('Broken beforeAll hook');
+    runner.addHook('beforeAll', () => {
+      throw error;
+    });
+    runner.addJourney(new Journey({ name: 'j1' }, () => step('step1', noop)));
+    const result = await runner.run({
+      wsEndpoint,
+      outfd: fs.openSync(dest, 'w'),
+    });
+    expect(result).toEqual({
```
Comment (andrewvc) [inline __tests__/core/runner.test.ts:160]: Can we add a second journey to this test? The hook should cause all journeys to fail, so we should assert it for > 1 journey

File: src/reporters/json.ts [hunk 2]
```diff
@@ -425,6 +423,9 @@ export default class JSONReporter extends BaseReporter {
           await gatherScreenshots(
             join(CACHE_PATH, 'screenshots'),
             async (step, data) => {
+              if (!data) {
```
Comment (andrewvc) [inline src/reporters/json.ts:426]: Can we add a note here mentioning that this is needed for fake journeys?

**Review by andrewvc — APPROVED** [review]
LGTM pending the minor changes mentioned

File: src/reporters/json.ts (hunk 2 above)
Comment (vigneshshanmugam) [reply src/reporters/json.ts:426]: I added this just incase, not really related to the commit.

File: __tests__/core/runner.test.ts (hunk 1 above)
Comment (vigneshshanmugam) [reply __tests__/core/runner.test.ts:160]: Yes good point, will add it.

**Pull request merged**